plugin/pkg/scheduler/core/extender_test.go

```diff
@@ -112,20 +113,81 @@ type FakeExtender struct {
 	filteredNodes    []*v1.Node
 }
 
+// selectVictimsOnNodeByExtender checks the given nodes->pods map with predicates on extender's side.
+// Returns:
+// 1. More victim pods (if any) amended by preemption phase of extender.
+// 2. Fits or not after preemption phase on extender's side.
+func (f *FakeExtender) selectVictimsOnNodeByExtender(
+	pod *v1.Pod,
+	node *v1.Node,
+	originVictims *schedulerapi.Victims,
+	pdbs []*policy.PodDisruptionBudget,
+) ([]*v1.Pod, int, bool) {
+	if fits, _ := f.doPredicate(pod, node); !fits {
+		return nil, 0, false
+	}
+
+	victims := []*v1.Pod{}
+	numViolatingVictim := 0
+
+	// TODO(harry): process preemption when we can have full nodeInfo here.
+	// And then we can also call util.FilterPodsWithPDBViolation() to calculate numViolatingVictim
+
+	return victims, numViolatingVictim, true
+}
```

Comment:
I once wrote a work preemption with `nodeInfo`, but finally found the `nodeInfo` is impossible to marshal for now. We may consider change this in the future.